src/axom/klee/GeometryOperatorsIO.cpp

```diff
@@ -0,0 +1,662 @@
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/klee/GeometryOperatorsIO.hpp"
+
+#include "axom/klee/Geometry.hpp"
+#include "axom/klee/GeometryOperators.hpp"
+#include "axom/klee/IOUtil.hpp"
+#include "axom/klee/KleeError.hpp"
+
+#include <functional>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace axom
+{
+namespace klee
+{
+namespace internal
+{
+namespace
+{
+using OpPtr = CompositeOperator::OpPtr;
+using OperatorParser =
+  std::function<OpPtr(const inlet::Container &,
+                      const TransformableGeometryProperties &)>;
+using internal::toDoubleVector;
+using primal::Point3D;
+using primal::Vector3D;
+using FieldSet = std::unordered_set<std::string>;
+
+/**
+ * Get the names of all the children in the given container.
+ *
+ * @param container the Container whose children to get
+ * @return the names of all the children
+ */
+std::unordered_set<std::string> getChildNames(const inlet::Container &container)
+{
+  std::unordered_set<std::string> allChildren;
+
+  std::vector<std::string> unexpectedNames = container.unexpectedNames();
+  allChildren.insert(unexpectedNames.begin(), unexpectedNames.end());
+
+  // Add 1 for the "/" separator
+  auto prefixLength = container.name().size() + 1;
+
+  for(auto &child : container.getChildContainers())
+  {
+    if(child.second->exists())
+    {
+      allChildren.insert(child.first.substr(prefixLength));
+    }
+  }
+
+  for(auto &child : container.getChildFields())
+  {
+    if(child.second->exists())
+    {
+      allChildren.insert(child.first.substr(prefixLength));
+    }
+  }
+
+  for(auto &child : container.getChildFunctions())
+  {
+    if(*child.second)
+    {
+      allChildren.insert(child.first.substr(prefixLength));
+    }
+  }
+
+  return allChildren;
+}
+
+/**
+ * Verify that a Container has the correct fields.
+ *
+ * \param containerToTest the Container to test
+ * \param name the name of the container. This must be one of its fields.
+ * \param additionalRequiredFields any additional required fields
+ * \param optionalFields any additional optional fields
+ */
+void verifyObjectFields(const inlet::Container &containerToTest,
+                        const std::string &name,
+                        const FieldSet &additionalRequiredFields,
+                        const FieldSet &optionalFields)
+{
+  std::unordered_set<std::string> requiredFields {additionalRequiredFields};
+  requiredFields.insert(name);
+
+  for(auto &requiredField : requiredFields)
+  {
+    if(!containerToTest.contains(requiredField))
+    {
+      std::string message = "Missing required parameter \"";
+      message += requiredField;
+      message += "\" for operator \"";
+      message += name;
+      message += '"';
+      throw KleeError({containerToTest.name(), message});
+    }
+  }
+
+  for(auto &child : getChildNames(containerToTest))
```

Comment:
This logic (and the accompanying `getChildNames`) makes me a little curious - was there a case where just using `unexpectedNames` was insufficient?  If I'm understanding this correctly `getChildNames` starts with the list of unexpected names and then adds the names of child fields, containers, and functions (all of which are by definition "expected").  The error here looks like it's thrown for anything in that set that isn't required or optional, which I would expect to match the set of unexpected names.